include/Foundation/NSDictionary+NSFileManagerExtensions.h

```diff
@@ -0,0 +1,38 @@
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#pragma once
+
+#include <Foundation/NSDictionary.h>
+
+@interface NSDictionary (NSFileManagerExtensions)
+- (NSDate*)fileCreationDate;
+- (BOOL)fileExtensionHidden;
+- (NSNumber*)fileGroupOwnerAccountID;
+- (NSString*)fileGroupOwnerAccountName;
+- (OSType)fileHFSCreatorCode;
+- (OSType)fileHFSTypeCode;
+- (BOOL)fileIsAppendOnly;
+- (BOOL)fileIsImmutable;
+- (NSDate*)fileModificationDate;
+- (NSNumber*)fileOwnerAccountID;
+- (NSString*)fileOwnerAccountName;
+- (NSUInteger)filePosixPermissions;
+- (unsigned long long int)fileSize;
+- (NSUInteger)fileSystemFileNumber;
+- (NSInteger)fileSystemNumber;
+- (NSString*)fileType;
+@end
```

Comment:
I'd be willing to bet these are all in `NSFileManager.h` on OS X: somebody including just NSFileManager would probably want to use them.